rosidl_typesupport_opensplice_c/resource/msg__type_support_c.cpp.template

```diff
@@ -69,30 +69,40 @@ ROSIDL_GET_TYPE_SUPPORT_FUNCTION(@(pkg), @(subfolder), @(msg))();
 have_not_included_primitive_arrays = True
 have_not_included_string = True
 }@
-@[for field in spec.fields]@
-@[if field.type.is_primitive_type()]@
-@[if field.type.is_array and have_not_included_primitive_arrays]@
+{
+from collections import OrderedDict
+field_types = {}
+for field in spec.fields:
+    if not field.type.is_primitive_type():
+        full_field = field.type.pkg_name + '__msg__' + field.type.type
```

Comment:
Correct me if I'm wrong, but I thought that `[Base]Type` didn't return the package name, so if two messages had the same base type name, but different packages, wouldn't the logic here be incorrect?